src/pages/Pools/index.tsx

```diff
@@ -206,8 +206,11 @@ export default function Pools() {
 
   const { loading: loadingUserLpPositions, data: userLpPairs } = useLPPairs(account || undefined)
 
-  const handleCurrencySelect = useCallback((token: any) => {
-    setFilterToken(token)
+  const handleCurrencySelect = useCallback((token: Currency) => {
+    // Since Token extends Currency we are checking address in useSwaprSinglelSidedStakeCampaigns
+    if (token.address) {
+      setFilterToken(token as Token)
+    }
```

Comment:
Just wondering if this can break something. Also, don't we always select a `Token` in pools page?

Reply:
As I commented above Token extends from Currency and have some more props. Its a big refactor.I dont think this will break since it was like this before :) with `any`